Google.PowerShell/Common/AuthenticateWithSdkCredentialsExecutor.cs

```diff
@@ -142,10 +121,16 @@ public void Initialize(ConfigurableHttpClient httpClient)
         /// </summary>
         public async Task<bool> RefreshTokenAsync(CancellationToken taskCancellationToken)
         {
-            var newToken = await GCloudWrapper.GetAccessToken();
-
-            Token = newToken;
-            return true;
+            await s_tokenLock.WaitAsync();
+            try
+            {
+                s_token = await GCloudWrapper.GetAccessToken(taskCancellationToken);
+                return true;
+            }
+            finally
+            {
+                s_tokenLock.Release();
+            }
```

Comment:
`return false;`? It seems strange to have a method that has a return type of `bool` and yet who only returns `true` or throws an exception.